File: src/Ekado.cpp
#include "../include/Ekado.hpp"

//Konstruktor Cheese
Ekado::Ekado(int _price,  ProductType _productType, ObjectType _objectType, char _objectSymbol)  {
    initializeProduct(_price, _productType, _objectType, _objectSymbol);
}

void InitResep() {
    //Inisialisasi resep:
    Ekado::getResep().add(EGG);
    Ekado::getResep().add(MEAT);
}

FarmProductType getResepwithIdx(int i) {
    return Ekado::getResep().get(i);
}

File: Java/src/main/java/engisubs/gameobject/product/side_product/Ekado.java
package main.java.engisubs.gameobject.product.side_product;

import java.util.LinkedList;

public class Ekado extends SideProduct{
    private static LinkedList<FarmProductType> resep;

    public Ekado(int _price){
        super.initializeProduct(_price, ProductType.SIDEPRODUCT, GameObjectType.EKADO, "Tiongkok's Special Ekado~~");
    }

    public static LinkedList<FarmProductType> getResep(){
        return resep;
    }

    public static FarmProductType getResepwithIdx(int index){
        return getResep().get(index);
    }

    public static void InitResep(){
        resep.add(FarmProductType.EGG);
        resep.add(FarmProductType.MEAT);
    }
}